src/generate_trigger.ts

```diff
@@ -0,0 +1,112 @@
+import { parse } from "./deps.ts";
+
+export type WorkflowRecord = {
+  "callback_id": string;
+};
+
+interface Inputs {
+  [key: string]: { value: string };
+}
+
+export type TriggerFileRecord = {
+  "type": string;
+  "name": string;
+  "description"?: string;
+  "workflow": WorkflowRecord;
+  "inputs"?: Inputs;
+};
+
+export type TriggersPayload = {
+  triggers: TriggerFileRecord[];
+};
+
+interface Response {
+  ok: boolean;
+  files: Array<string>;
+  error?: {
+    message: string;
+  } | null;
+}
+
+export const generateTriggerFile = async (
+  input: string[],
+): Promise<Response> => {
+  // Parse JSON payload
+  const source = parse(input).source as string;
+  if (!source) throw new Error("A source object needs to be defined");
+  const payload: TriggersPayload = JSON.parse(source);
+
+  // Create the triggers directory if it does not exist
+  const directory = "triggers";
+  try {
+    await Deno.stat(directory);
+  } catch (error) {
+    if (error instanceof Deno.errors.NotFound) { // create directory
+      await Deno.mkdir(directory);
+    } else {
+      throw error;
+    }
+  }
+
+  // Generate the files
+  const files: Array<string> = [];
+  await Promise.all(payload.triggers.map(async (tr: TriggerFileRecord) => {
+    const templateString = SlackTriggerTemplate(tr);
+    const filename = directory + `/${tr.workflow.callback_id}.ts`;
+
+    await Deno.writeTextFile(filename, templateString);
+    files.push(filename);
+  }));
+
+  return { ok: true, files: files };
+};
+
+const autogeneratedComment = () => {
+  const time = new Date();
+  return `/** This file was autogenerated on ${time.toDateString()}. **/`;
+};
+
+export const SlackTriggerTemplate = (tr: TriggerFileRecord) => {
+  const camelCase = tr.workflow.callback_id.toLowerCase().replace(
+    /[^a-zA-Z0-9]+(.)/g,
+    (_, chr) => chr.toUpperCase(),
+  );
+
+  let inputValues = ``;
+  // Add interactivity param to trigger inputs if present
+  if (tr.inputs) {
+    for (const prop in tr.inputs) {
+      if (tr.inputs[prop].value.includes("interactivity")) {
+        inputValues += `
+  ${prop}: {
+    value: "{{data.interactivity}}",
+  },
+`;
+      }
+    }
+  }
+
+  return `${autogeneratedComment()}
+import { Trigger } from "https://deno.land/x/deno_slack_api@1.2.0/types.ts";
```

Comment:
This is hardcoding the API version and it could be different than the API version they're using in their import map. Should we instead use `"import { Trigger } from "deno-slack-api/types.ts"`? This comes with it's own concern where the version of the API they're using might not support a version of the API library that exports `Trigger` that we'd need to resolve

Reply:
I wasn't sure if we could assume the user has the import_map.json file in their app defining "deno-slack-api" (maybe we can since our templates will have it) and as you mentioned, there's the possibility that future versions of that repo will have breaking changes :notsureif: